projects/developer/src/app/common/components/recipe-graph/component.ts

```diff
@@ -1180,4 +1180,17 @@ export class RecipeGraphComponent implements OnInit, OnChanges, AfterViewInit {
             this.update.next(true);
         }, 0);
     }
+
+    public hasConnection(filename, connections) {
+        const file_name = filename;
+        const connection_array = connections;
+        let has_connection = false;

```

Comment:
Not a big deal if you don't fix this, but this could be written more efficiently. A traditional for loop has an advantage here.

```js
for (let i = 0; i < connections.length; i++) {
    if (value.input_name === filename) {
        // no need to continue looping, breaks out of the loop and ends the function call
        // if you needed additional logic, use your variable and break here
        return true;
    }
}
return false;
```

Reply:
Sure, I can change it.